### PR TITLE
Removed the cache as essentially it is not needed in most cases

### PR DESCRIPTION
**Pull request by Skarlso**
Closes https://github.com/weaveworks/eksctl/issues/4616

Notes:

Most of the `Can*` methods where called right after `NewProviderForExistingCluster` which called `RefreshClusterStatus` by force. which means that the cluster info is available by the time Can* is called. In rare cases, this didn't happen, or could error and be ignored the error. In these cases, we just check if the status is set or not. The TTL didn't make actual sense, since the status wouldn't change.

Running integration tests on this branch...

https://github.com/weaveworks/eksctl-ci/actions/runs/1847254636

Outcome: SUCCESS 🟢 🟢

### CODE REVIEW

File: pkg/ctl/delete/cluster.go [hunk 1]
```diff
@@ -72,17 +70,13 @@ func doDeleteCluster(cmd *cmdutils.Cmd, force bool, disableNodegroupEviction boo
...
 	}
-	cmdutils.LogRegionAndVersionInfo(meta)
 
+	cmdutils.LogRegionAndVersionInfo(meta)
 	logger.Info("deleting EKS cluster %q", meta.Name)
 	if err := printer.LogObj(logger.Debug, "cfg.json = \\\n%s\n", cfg); err != nil {
 		return err
 	}
 
-	if ok, err := ctl.CanDelete(cfg); !ok {
```
Comment (Skarlso) [inline pkg/ctl/delete/cluster.go:82]: Can delete notes:

- `CanDelete`
    - Already called NewProvider before it
    - Probably can remove the TTL
    - Interesting `CanDelete` was called only once, but even so, it just errored out if Refresh failed. But that would have happened before it anyways

**Review by cPu1 — COMMENTED** [review]
One of the points to address in the issue was to ensure that if the local cluster state is being used after an update to the cluster's attributes, the updated attributes are reflected in the state. I'm curious if you found any such occurrences.

**Comment by Skarlso** [comment]
@cPu1 the only one I found that is questionable is `NewOpenIDConnectManager` and `GetCurrentClusterConfigForLogging`. But they seem to be working fine. Other than that, I didn't find a scenario where there was an update and cluster info was needed through `RefreshClusterStatusIfStale`. `SupportsManagedNodes` and `SupportsFargate` happen early in the process.

**Review by cPu1 — APPROVED** [review]

**Comment by Himangini** [comment]
The title of this PR doesn't look right, can you fix it ?

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**